### ISSUE TITLE
RecursionError using Latest Pages block with article pages

### ISSUE THREAD

**Issue by plascual** [coderedcorp/coderedcms#366]
#### Describe the bug
When using Latest Pages block on an article page, and selecting parent article landing page as source, if the current article page is in the listed latest pages, it crashes. The article landing page is also crashing.

#### Steps to reproduce
1. On an article page, use the Latest Pages block
2. Select parent page of current article as source
3. Select any number of pages to list, making sure the current page will be listed with the ordering specified in the parent’s LAYOUT tab.
4. RecursionError

#### Expected behavior
It should list all latest pages, including or excluding the current page.

**Comment by vsalvino** [comment]
Yes, this is a known bug. Thanks for reporting it with detailed steps to reproduce.

I'm not sure on the best way to solve this. One way would be to not allow selecting an ancestor of the current page in the latest pages block. Open to suggestions or a pull request if you have any ideas. Regardless, there should be better documentation with a warning about this behavior!

**Comment by plascual** [comment]
Ok, your post [#155](https://github.com/coderedcorp/coderedcms/pull/155) explains the known behavior.

IMHO, being able to select an ancestor is a must...

Would it be possible to simply exclude the current page from results, to avoid recursion?

**Comment by vsalvino** [comment]
Excluding the current page is not enough. If you select an ancestor (say a blog landing page) that page could render the sub-pages on it, as part of its content. Which then must load the current page in order to render the blog landing page. So there is simply no way to guarantee there won't be recursion - each page could be totally different based on what content the developer has put on it.

It's not just limited to ancestors. Say we have a Home Page which has two children, A and B. If A has a preview or latest pages of B, that will work. But if B also tries to do a preview or latest pages of A, you now have a recursion problem.

**Comment by plascual** [comment]
hmm...

Since page preview listing is not, strictly speaking, real content... maybe it should be excluded from content previewing. Does it make sense?

**Comment by rosslaird** [comment]
Has there been any resolution on this? I'm experiencing a recursion error when sibling article pages A and B each preview the other's page.

**Comment by vsalvino** [comment]
No resolution so far. The only possible solution I can see would be doing deep extensive validation of page preview blocks to prohibit the page from being saved when this will be an issue. But - there's no easy way to know if custom page types (especially with custom streamfields) are using the page preview block.
